ci/Jenkinsfile.windows

```diff
@@ -100,3 +100,10 @@ pipeline {
     cleanup { sh './scripts/clean-git.sh' }
   }
 }
+
+def getArch() {
+    def tokens = Thread.currentThread().getName().split('/')
+    for (def arch in ['x86_64', 'aarch64']) {
+      if (tokens.contains(arch)) { return arch }
+    }
+}
```

Comment:
This looks like it could be extracted to a shared place

Reply:
No, it can't. Because you can't use functions from libraries in the `environment` or `agent` contexts.

Reply:
Now, by library I mean what we load with the `library` directive, but in theor we could have an explicit:
```groovy
def common = load('common.groovy')
```
And then use that. It should work.

Reply:
No, I don't think I can use `load` outside of a `steps` and `script` contexts:
```
org.jenkinsci.plugins.workflow.steps.MissingContextVariableException: Required context class hudson.FilePath is missing
Perhaps you forgot to surround the code with a step that provides this, such as: node, dockerNode
```